Framework/DataObjects/inc/MantidDataObjects/WorkspaceCreation.h

```diff
@@ -158,13 +149,39 @@ std::unique_ptr<T> create(const P &parent, const IndexArg &indexArg,
       ws = detail::createConcreteHelper<T>();
     }
   }
+  return ws;
+}
 
+MANTID_DATAOBJECTS_DLL void
+initializeFromParent(const API::MatrixWorkspace &parent,
+                     API::MatrixWorkspace &ws);
+}
+
+template <class T, class P, class IndexArg, class HistArg,
+          class = typename std::enable_if<
+              std::is_base_of<API::MatrixWorkspace, P>::value>::type>
+std::unique_ptr<T> create(const P &parent, const IndexArg &indexArg,
+                          const HistArg &histArg) {
+  std::unique_ptr<T> ws = detail::createUninitialized<T, P>(parent);
   ws->initialize(indexArg, HistogramData::Histogram(histArg));
   detail::initializeFromParent(parent, *ws);
 
   return ws;
 }
 
+template <class T, class P, class IndexArg,
+          class = typename std::enable_if<
+              std::is_base_of<API::MatrixWorkspace, P>::value>::type>
+std::unique_ptr<T> create(const P &parent, const IndexArg &indexArg,
+                          const HistogramData::Histogram &histogram) {
```

Comment:
In the code above, the e and y data are removed only for EventWorkspaces, which don't use this data anyhow. However, if it is OK that
```c++
create<WorkspaceType>(parent, 2, parent.histogram(0));
```
works only if `WorkspaceType != EventWorkspace`, then I think I can completely revert the variants. Personally, I don't care either way, it is just that somebody might be surprised when EventWorkspaces suddenly cause runtime errors with something that works fine with Workspace2D.